src/subscriptions/__tests__/subscriptionsReducer-test.js

```diff
@@ -51,83 +53,35 @@ describe('subscriptionsReducer', () => {
     });
   });
 
-  test('on unrecognized action, returns input state unchanged', () => {
-    const prevState = deepFreeze({ hello: 'world' });
-
-    const newState = subscriptionsReducer(prevState, {});
-    expect(newState).toEqual(prevState);
-  });
-
   describe('EVENT_SUBSCRIPTION -> add', () => {
     test('if new subscriptions do not exist in state, add them', () => {
       const prevState = deepFreeze([]);
 
       const action = deepFreeze({
         type: EVENT_SUBSCRIPTION,
         op: 'add',
-        subscriptions: [
-          {
-            name: 'some stream',
-            stream_id: 1,
-          },
-          {
-            name: 'some other stream',
-            stream_id: 2,
-          },
-        ],
+        id: 1,
+        subscriptions: [sub1, sub2],
       });
 
-      const expectedState = [
-        {
-          name: 'some stream',
-          stream_id: 1,
-        },
-        {
-          name: 'some other stream',
-          stream_id: 2,
-        },
-      ];
+      const expectedState = [sub1, sub2];
 
       const newState = subscriptionsReducer(prevState, action);
 
       expect(newState).toEqual(expectedState);
     });
 
     test('if some of subscriptions already exist, do not add them', () => {
-      const prevState = deepFreeze([
-        {
-          color: 'red',
-          stream_id: 1,
-          name: 'some stream',
-        },
-      ]);
+      const prevState = deepFreeze([sub1]);
 
       const action = deepFreeze({
         type: EVENT_SUBSCRIPTION,
         op: 'add',
-        subscriptions: [
-          {
-            name: 'some stream',
-            stream_id: 1,
-          },
-          {
-            name: 'some other stream',
-            stream_id: 2,
-          },
-        ],
+        id: 1,
+        subscriptions: [sub1, sub2],
       });
 
-      const expectedState = [
-        {
-          color: 'red',
```

Comment:
Mm, I think that makes sense, but just to make sure:

> But that kind of substantive change in what behavior is expected should be its own commit, separate from the big mostly-mechanical change to make things well-typed, so that it can be described in its own commit message. Could come either before or after.

The "substantive change" is the one that this commit currently does, not an additional one you're requesting, right? (I gather this from, "It's probably fine that we keep the code simple and effectively assume this case doesn't happen".)

If it comes before the commit to start type checking, I think it's most straightforward to just add `color: 'red'` to the subscription in the action.

If it comes after the type-checking commit, I think it's more complicated. There needs to be a "color" property (and all the other required properties) *at* the type-checking commit. To emulate the current expectation at the type-checking commit (that the sub in state does not get clobbered), we need to change some properties of the sub in the action while keeping the ID the same. Then, after that commit, the new commit would bring it back to the normal, "boring" sub with the same ID and properties.

Best, I guess, if things are as boring as possible all the way through, so I'll maybe do it before the type-checking commit?